docs/concepts/ipns.md

```diff
@@ -5,19 +5,177 @@ description: Learn about the InterPlanetary Name System (IPNS) and how it can be
+
+#### How IPNS names relate to CIDs
+
+A CIDv1 can be used to represent both:
+
+- [Immutable content on IPFS](https://cid.ipfs.tech/#bafybeibml5uieyxa5tufngvg7fgwbkwvlsuntwbxgtskoqynbt7wlchmfm) (since the CID contains a multihash)
+- [IPNS name](https://cid.ipfs.tech/#k51qzi5uqu5dlvj2baxnqndepeb86cbk3ng7n3i46uzyxzyqj2xjonzllnv0v8) which corresponds to a libp2p public key.
+
+The following is a useful mental model for understanding the difference between the two:
 
-```shell
-/ipns/QmSrPmbaUKA3ZodhzPWZnpFgcPMFWF4QsxXbkWfEptTBJd
 ```
+IPFS = immutable *Pointer => content
+IPNS = **Pointer => content
+```
+
+IPNS names are essentially pointers (IPNS names) to pointers (IPFS CIDs) whereas IPFS CIDs are immutable (because they're derived from the content) pointers to content.
+
+### IPNS names are self-certifying
+
+IPNS names are self-certifying. This means that an IPNS record contains all the information necessary to certify its authenticity. IPNS achieves this using public and private key pairs:
+
+- Each IPNS name corresponds to a key pair
+- The IPNS name is a hash of the public key
+- The IPNS record contains the public key and signature, allowing anyone to verify that the record was signed by the private key holder.
+
+This self-certifying nature gives IPNS several benefits not preset in hierarchical and consensus systems such as DNS, and blockchain identifiers. Notably, IPNS records can come from anywhere, not just a particular service/system, and it is very fast and easy to confirm a record is authentic.
+
+### Common IPNS operations
+
+As a user or developer using IPNS for naming, there are three common operations worth understanding:
+
+- **Updating/Creating an IPNS record:** refers to the creation of an IPNS record and signing it with a private key.
+- **Publishing an IPNS record:** advertising the IPNS record so that other nodes can resolve it. Details depend on the transport.
+- **Resolving an IPNS name:** Resolving an IPNS name to a content path.
+
+### IPNS is transport agnostic
+
+The self-certifying nature of IPNS records means that they are not tied to a specific transport protocol. In practice, most IPFS implementations rely on the [**DHT**](dht.md) and **PubSub** to publish and resolve IPNS records.
+
+There are nuanced differences and trade-offs between the **DHT** and **PubSub** to be aware of.
+
+The main qualitative difference between the two is that IPNS over the DHT publishes and resolves to a global shared state, whereas IPNS over PubSub uses messaging over topics (where each IPNS name has a unique topic) to publish to and resolve from **interested peers**.
+
+The main implication of this difference is that IPNS operations (publishing and resolving) over the DHT can take longer than over PubSub, while potentially ensuring higher consistency (you resolve to the latest version).
+
+> **Note:** This trade-off is best explained by [CAP theorem](https://en.wikipedia.org/wiki/CAP_theorem).
```

Comment:
(feel free to ignore) I think we may want to use speciall callout blocks for notes like this, there is built-in support  documented at https://v1.vuepress.vuejs.org/guide/markdown.html#custom-containers

```suggestion
::: callout
This trade-off is best explained by [CAP theorem](https://en.wikipedia.org/wiki/CAP_theorem).
:::
```

Reply:
Generally, I agree though though the callout does make it a bit more prominent which in this specific case I think is unnecessary